<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SCREE</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
</style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

</head>

<body>


<div class="container-fluid main-container">

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="SCREE.html">SCREE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
          <a href="alignment.html">Alignment</a>
        </li>
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
            scRNA-seq
             
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu" role="menu">
            <li>
              <a href="RNA.html">Gene perturbation</a>
            </li>
            <li>
              <a href="ORF.html">ORF perturbation</a>
            </li>
            <li>
              <a href="enhancer.html">Enhancer perturbation</a>
            </li>
          </ul>
        </li>
        <li>
          <a href="ATAC.html">scATAC-seq</a>
        </li>
        <li>
          <a href="benchmark.html">Benchmark</a>
        </li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/HailinWei98/SCREE">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><strong>Benchmark</strong></h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SCREE/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210907code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210907)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210907code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210907)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHailinWei98SCREEtreedc5b2b27c0e24a9420f74a8c2a2e30c847fa6076targetblankdc5b2b2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/HailinWei98/SCREE/tree/dc5b2b27c0e24a9420f74a8c2a2e30c847fa6076" target="_blank">dc5b2b2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomHailinWei98SCREEtreedc5b2b27c0e24a9420f74a8c2a2e30c847fa6076targetblankdc5b2b2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/HailinWei98/SCREE/tree/dc5b2b27c0e24a9420f74a8c2a2e30c847fa6076" target="_blank">dc5b2b2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  data/workflow.png
    Untracked:  img/
    Untracked:  output/workflow.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCREE.Rmd</code>) and HTML (<code>docs/SCREE.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HailinWei98/SCREE/2c16336e7907134ed6b44f5232dcff325586db1c/docs/SCREE.html" target="_blank">2c16336</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HailinWei98/SCREE/blob/a2a5b0a1202aa108ab0750208f45cd85d8665385/analysis/SCREE.Rmd" target="_blank">a2a5b0a</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-23
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HailinWei98/SCREE/eeeebf3a496f7b67059be35871ae092db0f6d933/docs/SCREE.html" target="_blank">eeeebf3</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HailinWei98/SCREE/blob/94450d15ee4aa807a06ebcf1cb2096ca2b7f2b84/analysis/SCREE.Rmd" target="_blank">94450d1</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-23
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HailinWei98/SCREE/94450d15ee4aa807a06ebcf1cb2096ca2b7f2b84/docs/SCREE.html" target="_blank">94450d1</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-23
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HailinWei98/SCREE/518a5cacf3a46144afd24003ecf6a3a12ab7f80d/docs/SCREE.html" target="_blank">518a5ca</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HailinWei98/SCREE/blob/ab04ccf84f185bba13ce6d101eeae6c1ab1b290a/analysis/SCREE.Rmd" target="_blank">ab04ccf</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-22
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/HailinWei98/SCREE/855bd746a9419db56fd3d582b93134175f892b38/docs/SCREE.html" target="_blank">855bd74</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/HailinWei98/SCREE/blob/e582e5c4c363a978a4946ec77996a51d0ffa4b80/analysis/SCREE.Rmd" target="_blank">e582e5c</a>
</td>
<td>
HailinWei98
</td>
<td>
2021-09-22
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>
In this page, we provide the code for testing GSFA(v0.2.8) and SCEPTRE(V0.1.0). To estimate the time usage, we record the time before and after running the major model. To record the memory usage, we create a python script as follows and include the R code for running into the robjects.r.
</p>
<pre class="python"><code>from memory_profiler import profile
import rpy2.robjects as robjects
import os
    
@profile
def run():
    robjects.r('''{R code}''')

if __name__ == '__main__':
        run() 
</code></pre>
<h2>
<strong>GSFA Testing</strong>
</h2><br/>
<p>For GSFA testing, we find that its iteration step stop after finishing 800 iteration when we using a dataset of ~30K cells, 1.2K genes and 249 perturbed genes as input. </p>
<pre class="r"><code>library(Seurat)
library(GSFA)

#data preparation

mtx <- readRDS("perturb-CITE/control/RNA_QC.rds")
sg_lib <- read.table("perturb-CITE/control/sg_lib_all.txt", header = T)
sg_lib <- subset(sg_lib, cell %in% colnames(mtx))
scale.data <- GetAssayData(object = mtx, slot = "scale.data")

#define the function to convert the data frame to index matrix, derived from scmageck

frame2indmatrix <- function(bc_d, targetobj) {
    rnm = colnames(targetobj)
    cnm = unique(bc_d$gene)
    scalef = targetobj
    message(paste(length(unique(bc_d$cell)), "..."))
    message(paste(ncol(scalef), "..."))
    #rnm = rnm[!is.na(rnm)] #remove NA
    #rnm = rnm[rnm %in% colnames(scalef)]
    if (length(rnm) == 0) {
        stop("Cell names do not match in expression matrix and barcode.")
    }
    cnm = cnm[!is.na(cnm)]#remove NA
    ind_matrix = matrix(rep(0, length(rnm) * length(cnm)), nrow = length(rnm))
    rownames(ind_matrix) = rnm
    colnames(ind_matrix) = cnm
    row <- bc_d[, 'cell']
    col <- bc_d[, 'gene']
    test <- (row %in% rnm) & (col %in% cnm)
    idx <- as.matrix(data.frame(row[test], col[test]))
    ind_matrix[idx] <- 1
    return(ind_matrix)
}

G_mat <- frame2indmatrix(sg_lib, scale.data)

scale.data <- t(scale.data)
rm(list = c("mtx", "sg_lib"))


index <- which(colnames(G_mat) == "NTC")

#running GSFA

st<- Sys.time()
set.seed(50000)
fit0 <- fit_gsfa_multivar(Y = scale.data, G = G_mat, K = 20,
                          use_neg_control = TRUE,
                          neg_control_index = index,
                          prior_type = "mixture_normal", 
                          init.method = "svd",
                          niter = 1000, used_niter = 500,
                          verbose = T, return_samples = T)

set.seed(50000)
fit0 <- fit_gsfa_multivar(Y = scale.data, G = G_mat,
                          use_neg_control = TRUE,
                          neg_control_index = index,
                          prior_type = "mixture_normal", fit0 = fit0,
                          init.method = "svd",
                          niter = 1000, used_niter = 500,
                          verbose = T, return_samples = T)
et<- Sys.time()
print(et - st)
</code></pre>

<h2>
    <strong>SCEPTRE Testing</strong>
    </h2><br/>
    <p>For SCEPTRE testing, we execute the code with the parameter 'parallel = TRUE' to minimize the running time. However, we find that it will raise the warning message as follows when we test a dataset of ~15K cells, 1.2K genes and 4 perturbed genes. </p>
<pre class = "bash"><code>In mclapply(argsList, FUN, mc.preschedule = preschedule, mc.set.seed = set.seed,  : scheduled cores 80, 87, 97, 99, 103, 107, 108, 109, 111, 120, 126, 129, 132, 151, 154, 159, 176 did not deliver results, all values of the jobs will be affected.</code></pre>
<p>Due to this message, we don't test SCEPTRE with datasets including more cells. Additionally, while we execute the code with the parameter 'parallel = FALSE' recently, we find that the dataset used before can be analyzed by SCEPTRE without any warning message. We also test the new function in SCEPTRE v0.2.0 and find that the function can analyze datasets including more than 75K cells with less memory usage and more time usage, compared to the function in SCREE. We show the code for testing SCEPTRE v0.1.0 below.</p>
<p>To make sure the dimension of datasets for all methods is the same, we remove the build-in quality control step in the SCEPTRE function. The new functions are as follows.</p>
<pre class = "r"><code>run_sceptre_high_moi2 <- function(gene_matrix, combined_perturbation_matrix, covariate_matrix, gene_gRNA_group_pairs, side = "both", storage_dir = tempdir(), regularization_amount = 0.1, B = 1000, full_output = FALSE, parallel = TRUE, seed = 4) {
    ##################
    # DEFINE CONSTANTS
    ##################
    cat(paste0("Check ", storage_dir, "/logs for more detailed status updates.\n"))
  
    #############################
    # BASIC PROCESSING AND CHECKS
    #############################
    cat("Running checks and setting up directory structure. ")
  
    # 0. Set up parallel, fst, offsite directory structure, pod sizes
    dirs <- initialize_directories(storage_location = storage_dir)
    if (parallel) {
      n_cores <- parallel::detectCores()
      doParallel::registerDoParallel(cores = n_cores)
      foreach_funct <- foreach::`%dopar%`
    } else {
      n_cores <- 1L
      foreach_funct <- foreach::`%do%`
    }
    # 1. threshold gRNA_matrix (using threshold = 3, for now) if necessary
    gRNA_matrix <- combined_perturbation_matrix
  
    # 2. Ensure that cell barcodes coincide across gene and perturbation matrices
    if  (!identical(colnames(gRNA_matrix), colnames(gene_matrix)) || !identical(row.names(covariate_matrix), colnames(gene_matrix))) {
      stop("The column names of `gene_matrix` and `gRNA_matrix` and the row names of `covariate matrix` should be the cell barcodes. Ensure that the cell barcodes are present and identical across these matrices/data frames.")
    }
    if (is.null(row.names(gRNA_matrix)) || is.null(row.names(gene_matrix))) {
      stop("The row names of `gene_matrix` and `gRNA_matrix` should contain the gene IDs and gRNA IDs, respectively.")
    }
  
    # 4. Make sure genes/gRNAs in the data frame are actually a part of the expression matrices; ensure all rows distinct
    if ("gRNA_group" %in% colnames(gene_gRNA_group_pairs)) {
      gene_gRNA_group_pairs <- dplyr::rename(gene_gRNA_group_pairs, gRNA_id = gRNA_group)
    }
    if (!all(c("gene_id", "gRNA_id") %in% colnames(gene_gRNA_group_pairs))) stop("The columns `gene_id` and `gRNA_id` must be present in the `gene_gRNA_group_pairs` data frame.")
    abs_genes <- gene_gRNA_group_pairs$gene_id[!(gene_gRNA_group_pairs$gene_id %in% row.names(gene_matrix))]
    abs_gRNAs <- gene_gRNA_group_pairs$gRNA_id[!(gene_gRNA_group_pairs$gRNA_id %in% row.names(gRNA_matrix))]
    if (length(abs_genes) >= 1) {
      msg <- paste0("The genes `", paste0(abs_genes, collapse = ", "), "' are present in the `gene_gRNA_group_pairs` data frame but not in the `gene_matrix` matrix. Either remove these genes from `gene_gRNA_group_pairs,` or add these genes to `gene_matrix`.")
      stop(msg)
    }
    if (length(abs_gRNAs) >= 1) {
      msg <- paste0("The perturbations `", paste0(abs_gRNAs, collapse = ", "), "' are present in the `gene_gRNA_group_pairs` data frame but not in the `gRNA_matrix` matrix. Either remove these perturbations from `gene_gRNA_group_pairs,` or add these perturbations to `gRNA_matrix`.")
      stop(msg)
    }
    gene_gRNA_group_pairs <- gene_gRNA_group_pairs %>% dplyr::distinct()
    # 5. Set the pods
    n_genes <- length(unique(gene_gRNA_group_pairs$gene_id))
    n_gRNAs <- length(unique(gene_gRNA_group_pairs$gRNA_id))
    n_pairs <- nrow(gene_gRNA_group_pairs)
    pod_sizes <- c(gene = ceiling(n_genes/(2 * n_cores)),
                   gRNA = ceiling(n_gRNAs/(2 * n_cores)),
                   pair = ceiling(n_pairs/(2 * n_cores)))
    cat(crayon::green(' \u2713\n'))
    # 6. If number of genes is small, set regularization_amount to 0
    if (n_genes < 50) regularization_amount <- 0
  
    ##############
    # METHOD START
    ##############
    # create file dictionaries
    dicts <- suppressMessages(create_and_store_dictionaries(gene_gRNA_group_pairs,
                                           dirs[["gene_precomp_dir"]],
                                           dirs[["gRNA_precomp_dir"]],
                                           dirs[["results_dir"]],
                                           pod_sizes))
    cat("Running gene precomputations. ")
    # run first round of gene precomputations
    foreach_funct(foreach::foreach(pod_id = seq(1, dicts[["gene"]])),
                  run_gene_precomputation_at_scale_round_1(pod_id = pod_id,
                                                           gene_precomp_dir = dirs[["gene_precomp_dir"]],
                                                           gene_matrix = gene_matrix,
                                                           covariate_matrix = covariate_matrix,
                                                           regularization_amount = regularization_amount,
                                                           log_dir = dirs[["log_dir"]]))
    cat(crayon::green(' \u2713\n'))
  
    # regularize thetas
    regularize_gene_sizes_at_scale(gene_precomp_dir = dirs[["gene_precomp_dir"]],
                                   regularization_amount = regularization_amount,
                                   log_dir = dirs[["log_dir"]])
  
    # run second round of gene precomputations
    foreach_funct(foreach::foreach(pod_id = seq(1, dicts[["gene"]])),
                  run_gene_precomputation_at_scale_round_2(pod_id = pod_id,
                                                           gene_precomp_dir = dirs[["gene_precomp_dir"]],
                                                           gene_matrix = gene_matrix,
                                                           covariate_matrix = covariate_matrix,
                                                           regularization_amount = regularization_amount,
                                                           log_dir = dirs[["log_dir"]]))
  
    # run gRNA precomputations
    cat("Running perturbation precomputations. ")
    foreach_funct(foreach::foreach(pod_id = seq(1, dicts[["gRNA"]])),
                  run_gRNA_precomputation_at_scale(pod_id = pod_id,
                                                   gRNA_precomp_dir = dirs[["gRNA_precomp_dir"]],
                                                   combined_perturbation_matrix = combined_perturbation_matrix,
                                                   covariate_matrix = covariate_matrix,
                                                   log_dir = dirs[["log_dir"]],
                                                   B = B,
                                                   seed = seed))
    cat(crayon::green(' \u2713\n'))
  
    # run at-scale analysis
    cat("Running perturbation-to-gene association tests.")
    foreach_funct(foreach::foreach(pod_id = seq(1, dicts[["pairs"]])),
                  run_gRNA_gene_pair_analysis_at_scale(pod_id = pod_id,
                                                       gene_precomp_dir = dirs[["gene_precomp_dir"]],
                                                       gRNA_precomp_dir = dirs[["gRNA_precomp_dir"]],
                                                       results_dir = dirs[["results_dir"]],
                                                       log_dir = dirs[["log_dir"]],
                                                       gene_matrix = gene_matrix,
                                                       combined_perturbation_matrix = combined_perturbation_matrix,
                                                       covariate_matrix = covariate_matrix,
                                                       regularization_amount = regularization_amount,
                                                       side = side,
                                                       B = B,
                                                       full_output))
    cat(crayon::green(' \u2713\n'))
    # collect results
    cat("Collecting and returning results. ")
    results <- collect_results(results_dir = dirs[["results_dir"]], gene_gRNA_group_pairs)
    cat(crayon::green(' \u2713\n'))
    return(results)
  }
  </code></pre>
<p>The code for SCEPTRE testing. The function <code>frame2indmatrix</code> is the same as the function mentioned in GSFA testing part.</p>
<pre class="r"><code>library(dplyr)
library(sceptre)
library(Seurat)

#data preparation

sg_lib<- read.table("GSE90546/GSM2406677_10X005/sg_lib_all.txt", header=T)
data<- readRDS("GSE90546/GSM2406677_10X005/RNA.rds")

#quality control

data<- subset(data, sgRNA_num != 0 & 
                nFeature_RNA >= 200 & 
                nCount_RNA >= 1000 &
                percent.mt <= 10)
data_qc <- CreateSeuratObject(count = data@assays$RNA@counts, min.cells = 0.01 * ncol(data))
data_qc[["percent.mt"]] <- PercentageFeatureSet(data_qc, pattern = "^MT-")
sg_num <- subset(data$sgRNA_num, names(data$sgRNA_num) %in% colnames(data_qc))
perturbations<- subset(data$perturbations, names(data$perturbations) %in% colnames(data_qc))

data_qc[["log_UMI"]]<- log(data_qc$nCount_RNA)

sg_lib <- subset(sg_lib, cell %in% colnames(data_qc))
sg_lib$gene <- gsub("NTC", "non-targeting", sg_lib$gene)
sg_lib$barcode <- gsub("NTC_", "non-targeting_", sg_lib$barcode)

#Create gene-sgRNA pairs

l <- length(unique(sg_lib$gene))
gene_id<- rep(rownames(data_qc), times = l)
gRNA_id<- rep(unique(sg_lib$gene), each = nrow(data_qc))
gene_gRNA_pairs<- data.frame(gene_id = gene_id, gRNA_group = gRNA_id)

#Generate perturbation index matrix

perturbation_matrix <- t(frame2indmatrix(sg_lib, data_qc))

#Generate covariate matrix

covariate_matrix <- data_qc[[c("log_UMI", "percent.mt")]]

#Get sparse matrix

expression_matrix <- data_qc@assays$RNA@counts

sg_lib <- unique(sg_lib[, c("barcode", "gene")])
grna_group_df <- data.frame(grna_id = sg_lib$barcode, grna_group = sg_lib$gene)

rm(names = "l","sg_lib","data","data_qc","sg_num","perturbations")

dir.create("comparison_results/sceptre_results")

#Running SCEPTRE

st<- Sys.time()
result <- run_sceptre_high_moi2(storage_dir = "comparison_results/sceptre_results",
                                gene_matrix = expression_matrix,
                                combined_perturbation_matrix = perturbation_matrix,
                                covariate_matrix = covariate_matrix,
                                gene_gRNA_group_pairs = gene_gRNA_pairs,
                                side = "both", 
                                B = 2000,
                                parallel = T)
et<- Sys.time()
print(et-st)
    </code></pre>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        whisker_0.4       knitr_1.33        magrittr_2.0.1   
 [5] R6_2.5.0          rlang_0.4.11      fansi_0.5.0       stringr_1.4.0    
 [9] tools_4.0.2       xfun_0.25         utf8_1.2.2        git2r_0.28.0     
[13] jquerylib_0.1.4   htmltools_0.5.1.1 ellipsis_0.3.2    rprojroot_2.0.2  
[17] yaml_2.2.1        digest_0.6.27     tibble_3.1.3      lifecycle_1.0.0  
[21] crayon_1.4.1      later_1.2.0       sass_0.4.0        vctrs_0.3.8      
[25] promises_1.2.0.1  fs_1.5.0          glue_1.4.2        evaluate_0.14    
[29] rmarkdown_2.10    stringi_1.7.3     bslib_0.2.5.1     compiler_4.0.2   
[33] pillar_1.6.2      jsonlite_1.7.2    httpuv_1.6.1      pkgconfig_2.0.3  </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<script>

  // add bootstrap table styles to pandoc tables
  function bootstrapStylePandocTables() {
    $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
  }
  $(document).ready(function () {
    bootstrapStylePandocTables();
  });
  
  
  </script>

</div>
</div>

</div>


</body>
</html>
